### PR TITLE
Use modern requirements

### PR DESCRIPTION
**Pull request by mwojcikowski**
Finally some packages were retired, including OB 2.4.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,15 +4,11 @@ cache: pip
...
-        - ODDT_TOOLKIT=ob CONDA_PY=2.7 MINIMAL_REQ=1
-        - ODDT_TOOLKIT=rdk CONDA_PY=2.7  MINIMAL_REQ=1
-        - ODDT_TOOLKIT=ob CONDA_PY=3.5 MINIMAL_REQ=1
-        - ODDT_TOOLKIT=rdk CONDA_PY=3.5  MINIMAL_REQ=1
+        - ODDT_TOOLKIT=ob CONDA_PY=3.6 MINIMAL_REQ=1
+        - ODDT_TOOLKIT=rdk CONDA_PY=3.6  MINIMAL_REQ=1
         # Recent versions
-        - ODDT_TOOLKIT=ob CONDA_PY=2.7
         - ODDT_TOOLKIT=ob CONDA_PY=3.6
         - ODDT_TOOLKIT=ob CONDA_PY=3.7
```
Comment (marta-sd) [inline .travis.yml:11]: will it explode if we add py3.8?

File: README.md [hunk 2]
```diff
@@ -15,12 +15,12 @@ Open Drug Discovery Toolkit (ODDT) is modular and comprehensive toolkit for use
...
 ## Requrements
-  * Python 2.7 or 3.5+
-  * OpenBabel (2.3.2+) or/and RDKit (2016.03)
-  * Numpy (1.11+)
-  * Scipy (0.17+)
+  * Python 3.6+
+  * OpenBabel (3.0+) or/and RDKit (2018.03+)
+  * Numpy (1.12+)
+  * Scipy (0.19+)
   * Sklearn (0.18+)
```
Comment (marta-sd) [inline README.md:22]: I can also see some `if sklearn_version >= '0.19'`, maybe we should get rid 0.18?

File: README.md [hunk 3]
```diff
@@ -15,12 +15,12 @@ Open Drug Discovery Toolkit (ODDT) is modular and comprehensive toolkit for use
...
-  * joblib (0.9.4+)
+  * joblib (0.10+)
   * pandas (0.19.2+)
   * Skimage (0.12.3+) (optional, only for surface generation)
```
Comment (marta-sd) [inline README.md:25]: and same for skimage: `if LooseVersion(skimage_version) >= LooseVersion('0.13')`

File: .travis.yml (hunk 1 above)
Comment (mwojcikowski) [reply .travis.yml:11]: It probably will, so I'll do it.

File: README.md (hunk 3 above)
Comment (mwojcikowski) [reply README.md:25]: That one is still relevant, but we may bump the requirements to 0.13 instead.

**Pull request merged**